source/include/Date.h

```diff
@@ -54,7 +54,7 @@ class Date {
         Date(Timestamp timestamp, Int32 timeZoneOffset); // this API is conceptually wrong; tzo should be computed from UTC date
         Date(Timestamp timestamp, bool isUTC = false);
         ~Date();
-        static Int32 getLocaletimeZone();
+        static Int32 getLocaletimeZone(Int64 utcTime = 0);
```

Comment:
Should we really need to support default value of 0 here? The one place the default value is being used seems to be "oid Format(SubString *format, Int32 count, StaticTypeAndData arguments[], StringRef buffer)" in NumericString.cpp. In that, if timezone is not 0, should we try to use again the passed timestamp from the argument? That may need a bit of refactoring.

If there is a need for this, I would add a second boolean "isUTC" argument as done in other places (instead of default value of 0) to make this clear and consistent at the callsites. If this function is used incorrectly without passing argument, I wonder whether some tests that use the exact epoch as timestamp (such that this is 0) may be failing?